File: web/src/api/mine/mine_websites.js
import service from '@/utils/request'

export const createMineWebsites = (data) => {
  return service({
    url: '/mineWebsites/createMineWebsites',
    method: 'post',
    data
  })
}

export const deleteMineWebsites = (params) => {
  return service({
    url: '/mineWebsites/deleteMineWebsites',
    method: 'delete',
    params
  })
}

export const deleteMineWebsitesByIds = (params) => {
  return service({
    url: '/mineWebsites/deleteMineWebsitesByIds',
    method: 'delete',
    params
  })
}

export const updateMineWebsites = (data) => {
  return service({
    url: '/mineWebsites/updateMineWebsites',
    method: 'put',
    data
  })
}

export const findMineWebsites = (params) => {
  return service({
    url: '/mineWebsites/findMineWebsites',
    method: 'get',
    params
  })
}

export const getMineWebsitesList = (params) => {
  return service({
    url: '/mineWebsites/getMineWebsitesList',
    method: 'get',
    params
  })
}

File: server/router/mine/mine_websites.go
package mine

import (
	"github.com/gin-gonic/gin"
	"github.com/springbear2020/self-hub/server/middleware"
)

type MineWebsitesRouter struct{}

func (s *MineWebsitesRouter) InitMineWebsitesRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	mineWebsitesRouter := Router.Group("mineWebsites").Use(middleware.OperationRecord())
	mineWebsitesRouterWithoutRecord := Router.Group("mineWebsites")
	mineWebsitesRouterWithoutAuth := PublicRouter.Group("mineWebsites")
	{
		mineWebsitesRouter.POST("createMineWebsites", mineWebsitesApi.CreateMineWebsites)
		mineWebsitesRouter.DELETE("deleteMineWebsites", mineWebsitesApi.DeleteMineWebsites)
		mineWebsitesRouter.DELETE("deleteMineWebsitesByIds", mineWebsitesApi.DeleteMineWebsitesByIds)
		mineWebsitesRouter.PUT("updateMineWebsites", mineWebsitesApi.UpdateMineWebsites)
	}
	{
		mineWebsitesRouterWithoutRecord.GET("findMineWebsites", mineWebsitesApi.FindMineWebsites)
		mineWebsitesRouterWithoutRecord.GET("getMineWebsitesList", mineWebsitesApi.GetMineWebsitesList)
	}
	{
		_ = mineWebsitesRouterWithoutAuth
	}
}
